src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -842,11 +860,40 @@ where
             is_new_user_key = false;
         }
 
-        // Don't allow two SSTs to share same user key
-        sst_builder
-            .add_full_key(iter_key, value, is_new_user_key)
-            .verbose_instrument_await("add_full_key")
-            .await?;
+        // May drop stale columns
+        let check_table_id = last_key.user_key.table_id.table_id;
+        let mut is_value_rewritten = false;
+        if let HummockValue::Put(v) = value
+            && let Some(object_id) = object_id
+            && let Some(block_id) = block_id
+            && skip_schema_check.get(&object_id).map(|prev_block_id|*prev_block_id != block_id).unwrap_or(true)
```

Comment:
nits: `skip_schema_check` skip_schema_check represents the optimization of skip? 

IMO， It will be clearer like
```rust
!skip_schema_check.get(&object_id).map(|prev_block_id|*prev_block_id == block_id).unwrap_or(false)
```

feel free to ignore these comments.

Reply:
fixed